### PR TITLE
refactor(amazonq): add BaseConnector Abstract class for common logic

### PR DESCRIPTION
**Pull request by ivikash**
Create an abstract class that can be used by all connectors.

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/amazonq/webview/ui/apps/BaseConnector.ts [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+            link,
+            tabType: this.getTabType(),
+        })
+    }
+
+    followUpClicked = (tabID: string, messageId: string, followUp: ChatItemAction): void => {
+        /**
+         * We've pressed on a followup button and should start watching that round trip telemetry
+         */
+        this.sendMessageToExtension({
```
Comment (jpinkney-aws) [inline packages/core/src/amazonq/webview/ui/apps/BaseConnector.ts:71]: I'd probably move these out of here for now and into the cwchatconnector. I haven't checked whether or not the entire chat round trip telemetry flow works for gumby/feature dev yet

File: packages/core/src/amazonq/webview/ui/apps/BaseConnector.ts
Comment (justinmk3) [inline packages/core/src/amazonq/webview/ui/apps/BaseConnector.ts:1]: file name should be camelcase (baseConnector.ts)

**Review by justinmk3 — APPROVED** [review]
Nice! Are there any tests that could be de-duplicated too?

File: packages/core/src/amazonq/webview/ui/apps/BaseConnector.ts (hunk 1 above)
Comment (ivikash) [reply packages/core/src/amazonq/webview/ui/apps/BaseConnector.ts:71]: Discussed Offline - `amazonq_chatRoundTrip` is not getting emitted for `/dev`, so this is good to go.

**Review by jpinkney-aws — APPROVED** [review]

**Pull request merged**